packages/cli/src/cli/index.ts

```diff
@@ -198,6 +197,108 @@ $$$$$$\\   $$$$$$$\\  $$\\  $$$$$$\\   $$$$$$$ |$$\\  $$\\  $$\\  $$$$$$\\  $$$$
       await detectExtensions(options);
     });
 
+  // eslint-disable-next-line turbo/no-undeclared-env-vars
+  if (process.env.THIRDWEB_CLI_SKIP_INTRO !== "true") {
+    const versionSpinner = spinner("Checking for updates...");
+    await import("update-notifier").then(
+      async ({ default: updateNotifier }) => {
+        const notifier = updateNotifier({
+          pkg,
+          shouldNotifyInNpmScript: true,
+          // check every time while we're still building the CLI
+          updateCheckInterval: 0,
+        });
+
+        const versionInfo = await notifier.fetchInfo();
+        versionSpinner.stop();
+
+        if (versionInfo.type !== "latest") {
+          const res = await prompts({
+            type: "toggle",
+            name: "upgrade",
+            message: `A new version of the CLI is available. Would you like to upgrade?`,
+            initial: true,
+            active: "yes",
+            inactive: "no",
+          });
+
+          if (res.upgrade) {
+            const updateSpinner = spinner(
+              `Upgrading CLI to version ${versionInfo.latest}...`,
+            );
+
+            // Check if thirdweb package is already installed in local package.json
+            const localPackages: string[] = await new Promise((resolve) => {
+              exec("npm ls --depth=0", (err, stdout) => {
+                var packages: string[] = [];
+                packages = stdout.split("\n");
+                packages = packages.filter(function (item) {
+                  if (item.match(/^├──.+/g) !== null) {
+                    return true;
+                  }
+                  if (item.match(/^└──.+/g) !== null) {
+                    return true;
+                  }
+                  return undefined;
+                });
+                packages = packages
+                  .map(function (item) {
+                    if (item.match(/^├──.+/g) !== null) {
+                      return item.replace(/^├──\s/g, "");
+                    }
+                    if (item.match(/^└──.+/g) !== null) {
+                      return item.replace(/^└──\s/g, "");
+                    }
+                  })
+                  .filter((item) => !!item) as string[];
+                resolve(packages);
+              });
+            });
+
+            const isLocal = localPackages.find((packageName) =>
+              packageName.includes("thirdweb@"),
+            );
+            const command = isLocal ? "npm i thirdweb" : "npm i -g thirdweb";
```

Comment:
did you try it on replit?

Reply:
what if I use `yarn` in my project, or `pnpm` ?

Reply:
Just tried it out on repl.it, looks like everything works (global install working)